lib/ClangImporter/ImportName.cpp

```diff
@@ -2201,6 +2201,38 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
       return importNameImpl(classTemplateSpecDecl->getSpecializedTemplate(),
                             version, givenName);
     if (!isa<clang::ClassTemplatePartialSpecializationDecl>(D)) {
+      auto getSwiftBuiltinTypeName =
+          [&](const clang::BuiltinType *builtin) -> std::optional<StringRef> {
+        Type swiftType = nullptr;
+        switch (builtin->getKind()) {
+        case clang::BuiltinType::Void:
+          swiftType = swiftCtx.getNamedSwiftType(swiftCtx.getStdlibModule(),
+                                                 "Void");
+          break;
+#define MAP_BUILTIN_TYPE(CLANG_BUILTIN_KIND, SWIFT_TYPE_NAME)                  \
+        case clang::BuiltinType::CLANG_BUILTIN_KIND:                           \
+          swiftType = swiftCtx.getNamedSwiftType(swiftCtx.getStdlibModule(),   \
+                                                 #SWIFT_TYPE_NAME);            \
+          break;
+#define MAP_BUILTIN_CCHAR_TYPE(CLANG_BUILTIN_KIND, SWIFT_TYPE_NAME)            \
+        case clang::BuiltinType::CLANG_BUILTIN_KIND:                           \
+          swiftType = swiftCtx.getNamedSwiftType(swiftCtx.getStdlibModule(),   \
+                                                 #SWIFT_TYPE_NAME);            \
+          break;
+#include "swift/ClangImporter/BuiltinMappedTypes.def"
+        default:
+          break;
+        }
+
+        if (swiftType) {
+          if (auto nominal =
+                  dyn_cast<NominalType>(swiftType->getCanonicalType())) {
```

Comment:
`swiftType->getAs<NominalType>()` is simpler

Reply:
Indeed, thanks!